src/vs/workbench/contrib/welcome/gettingStarted/browser/gettingStarted.ts

```diff
@@ -264,7 +264,14 @@ export class GettingStartedPage extends EditorPane {
 							}
 							const taskToRun = assertIsDefined(this.currentCategory?.content.items.find(task => task.id === argument));
 							if (taskToRun.button.command) {
-								this.commandService.executeCommand(taskToRun.button.command);
+								if (taskToRun.button.sideBySide) {
+									// todo: Only split when current layout is 1-column
+									this.commandService.executeCommand('workbench.action.editorLayoutTwoColumns')
+										.then(() => this.commandService.executeCommand('workbench.action.focusNextGroup'))
+										.then(() => this.commandService.executeCommand(taskToRun.button.command));
```

Comment:
Another todo: There is a TS error here that `command` might be `undefined`, which is odd as it is within the `if` and the same line works 2 lines below.

Reply:
TS discards the inferred constraints once you're in a new function scope (the `.then` block could happen a long time after the `if`, in the mean time a different task could change the definedness). Not an actual issue here, but TS doesn't know that.